db/blob/db_blob_compaction_test.cc

```diff
@@ -0,0 +1,293 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "db/blob/blob_index.h"
+#include "db/db_test_util.h"
+#include "port/stack_trace.h"
+#include "test_util/sync_point.h"
+#include "utilities/fault_injection_env.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class DBBlobCompactionTest : public DBTestBase {
+ public:
+  DBBlobCompactionTest()
+      : DBTestBase("/db_blob_compaction_test", /*env_do_fsync=*/false) {}
+};
+
+namespace {
+
+class DummyStackedBlobDbCompactionFilter : public CompactionFilter {
+ public:
+  DummyStackedBlobDbCompactionFilter() = default;
+  const char* Name() const override {
+    return "rocksdb.compaction.filter.dummy.stacked.blobdb";
+  }
+  bool IsStackedBlobDbCompactionFilter() const override { return true; }
+};
+
+class FilterByKeyLength : public CompactionFilter {
+ public:
+  FilterByKeyLength(size_t len) : length_threshold_(len) {}
```

Comment:
Minor: these single-argument constructors could be marked `explicit` to make our linter happy.